### PR TITLE
Draft: Branch to cleanup ssc/api ready for merge.

### PR DESCRIPTION
**Pull request by isolatedinformation**
This message body will be used to track all the small changes made, since there will be quite a few.
- Suppress PyMatching and inaccesible file warnings when running the simulations.
- Fix incorrectly generated stabilizers of the rectangular versions of the rotated planar code. 
- Make MeasurementSample use the weight of the logical operator instead of code distance and support multiple logical operator measurements. 
- Fix the stabilizers for the rotated planar code for reactangular and XZZX versions.

Update: 25-10-2023 19:30, after all these changes I ran the previously obtained threshold plots for the rotated planar code, and they are giving the right thresholds. The rectangular codes however are not being processed currently. I will post more updates in the next update. 

Update: 26-10-2023 17:30 After fixing the stabilizers of the rotated planar code for the non-square versions, we have now how threshold plots that agree with literature upto statistical accuracies.

### CODE REVIEW

**Comment by antalszava** [comment]
Hi @isolatedinformation, I know that this PR is in work in progress, but I've come across a failure and thought of mentioning it as FYI. Running `examples/ler_decreases.py` seems to fail with:

```py
Traceback (most recent call last):
  File "/plaquette/examples/ler_decreases.py", line 66, in <module>
    results = MeasurementSample.from_code_and_raw_results(ssc, raw, erasure, num_rounds)
  File "/plaquette/src/plaquette/device/__init__.py", line 115, in from_code_and_raw_results
    logop_wts = [code.distances[j] for j in logop_idxs]
  File "/plaquette/src/plaquette/device/__init__.py", line 115, in <listcomp>
    logop_wts = [code.distances[j] for j in logop_idxs]
IndexError: index 2 is out of bounds for axis 0 with size 2
```

**Comment by isolatedinformation** [comment]
Thanks for the report. Yes, there is a bug because the signature of the changed. You now also have to give the logical_ops we have measured to process the measurement sample.

File: requirements.txt [hunk 1]
```diff
@@ -18,3 +18,4 @@ stim~=1.12
 toml>=0.10
 tqdm
 rustworkx
+igraph
```
[the remark this answers is not in the document]
Comment (isolatedinformation) [reply requirements.txt]: Nope, those dependencies were removed everywhere from the code. icecream is just a nice command line debugging tool.

File: src/plaquette/codes/utils.py
Comment (massarom) [inline src/plaquette/codes/utils.py:1]: If the functions in this module are utilities *for the developers*, then the entire module should be private `utils.py -> _utils.py` and the functions inside can be normal. I think mypy or some other type checkers complains about using private functions in a public module.

File: src/plaquette/decoders/__init__.py [hunk 2]
```diff
@@ -386,7 +386,17 @@ def __init__(
                 p = eff_error_probabilities[pauli_factor.name.lower()][
                     self._data_qubit_index_for_edge[edge_idx]
                 ]
-                edges_weights.append(-np.log(p / (1 - p)))
+
+                if p == 0.0:
+                    edges_weights.append(2**24 - 1)
+                elif p == 1.0:
+                    edges_weights.append(-(2**24) + 1)
```
Comment (massarom) [inline src/plaquette/decoders/__init__.py:393]: Why not `float('inf')`? Only the specific decoder wrappers should care about precise values.

File: src/plaquette/decoders/__init__.py [hunk 3]
```diff
@@ -450,7 +460,10 @@ def __init__(
                         # error probability greater than zero
                         ancilla = t.cast(NodeIndex, ancilla + code.num_data_qubits)
                         p = eff_error_probabilities["measurement"][ancilla]
-                    edges_weights.append(-np.log(p / (1 - p)))
+
+                    with warnings.catch_warnings():
+                        warnings.simplefilter("ignore")
+                        edges_weights.append(-np.log(p / (1 - p)))
```
Comment (massarom) [inline src/plaquette/decoders/__init__.py:466]: Why these are not treated the same as in-plane edges?

File: src/plaquette/device/__init__.py [hunk 4]
```diff
@@ -99,32 +100,38 @@ def from_code_and_raw_results(
...
         """
-        code_distance = code.distance
-
+        logop_idxs = [
+            i + code.num_logical_qubits if op == "Z" else i
+            for i, op in enumerate(logical_ops)
+        ]
+        # ic(logop_idxs)
+        logop_wts = [code.distances[j] for j in logop_idxs]
+        # ic(logop_wts)
```
Comment (massarom) [inline src/plaquette/device/__init__.py:116]: I'm sorry, but these two comments are too cryptic. Can you expand the acronyms?

File: src/plaquette/errors/common_models.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    _GateErrorMetadata,
+    _QubitErrorMetadata,
+)
+
+
+def generate_standard_circuit_noise(code: Code, p: float) -> ErrorData:
+    """Generate the standard depolarizing circuit noise model.
+
+    Args:
+        code: The `Code` object to define errors for.
```
Comment (massarom) [inline src/plaquette/errors/common_models.py:18]:
```suggestion
        code: the :class:`.Code` object to define errors for.
```

File: tests/test_regression_ler_decrease.py [hunk 6]
```diff
@@ -91,16 +91,14 @@ def get_logical_error_rate(
...
+class TestLERUnderThresholdPyMatching:
     """Test decrease of logical error rates for growing code size under threshold."""
 
     @pt.mark.slow
     @pt.mark.parametrize(
         "decoder_cls",
         [
             decoders.PyMatchingDecoder,
-            decoders.FusionBlossomDecoder,
-            decoders.UnionFindDecoder,
```
Comment (massarom) [inline tests/test_regression_ler_decrease.py:103]: No need to parametrise a variable that only ever has one parameter. (applies also to below)

**Review by massarom — CHANGES_REQUESTED** [review]
Besides the line-level comments, any reason why this

```
FAILED tests/integrationtests/generator_test.py::TestQECCircuitGenerator::test_generated_syndrome - IndexError: index 2 is out of bounds for axis 0 with size 2
```

fails? The other failing tests should be fixed in the docs branch, but this one is weird.

**Review by massarom — APPROVED** [review]
Good as long as the tests pass

**Pull request merged**